{
  "url": "https://api.github.com/repos/shap/shap/issues/720",
  "repository_url": "https://api.github.com/repos/shap/shap",
  "labels_url": "https://api.github.com/repos/shap/shap/issues/720/labels{/name}",
  "comments_url": "https://api.github.com/repos/shap/shap/issues/720/comments",
  "events_url": "https://api.github.com/repos/shap/shap/issues/720/events",
  "html_url": "https://github.com/shap/shap/issues/720",
  "id": 473739229,
  "node_id": "MDU6SXNzdWU0NzM3MzkyMjk=",
  "number": 720,
  "title": "Having issues with learning_phase_flags",
  "user": {
    "login": "svadams",
    "id": 8512966,
    "node_id": "MDQ6VXNlcjg1MTI5NjY=",
    "avatar_url": "https://avatars.githubusercontent.com/u/8512966?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/svadams",
    "html_url": "https://github.com/svadams",
    "followers_url": "https://api.github.com/users/svadams/followers",
    "following_url": "https://api.github.com/users/svadams/following{/other_user}",
    "gists_url": "https://api.github.com/users/svadams/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/svadams/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/svadams/subscriptions",
    "organizations_url": "https://api.github.com/users/svadams/orgs",
    "repos_url": "https://api.github.com/users/svadams/repos",
    "events_url": "https://api.github.com/users/svadams/events{/privacy}",
    "received_events_url": "https://api.github.com/users/svadams/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [

  ],
  "state": "open",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 0,
  "created_at": "2019-07-28T12:35:02Z",
  "updated_at": "2019-07-28T12:35:02Z",
  "closed_at": null,
  "author_association": "NONE",
  "active_lock_reason": null,
  "body": "Hi!\r\nFirst of all - great work on shap!\r\n\r\nI have encountered some issues with the auto detection of learning phase flags when using a keras model. I was trying to use the InceptionV3 pretrained model from keras with the shap Imagenet50 dataset but the code in deep_tf.py does not locate the correct Placeholder ops from the batch_normalization layers and this error is given:\r\n\r\ntensorflow.python.framework.errors_impl.InvalidArgumentError: You must feed a value for placeholder tensor 'batch_normalization_1/keras_learning_phase' with dtype bool\r\n\r\nI think this is because the deep_tf.py code has these conditions:\r\n\r\nif 'learning_phase' in op.name and op.type == \"Const\" and len(op.outputs[0].shape) == 0\r\n\r\nI can hack it to only look for op.type == \"Placeholder\" and get past this. I also tried passing a Placeholder tensor when creating the DeepExplainer object but got another error (self.learning_phase_flags does not exist).\r\n\r\nI don't specifically need shap to work with InceptionV3 as I was just exploring but thought it was worth reporting.\r\n\r\n",
  "closed_by": null,
  "reactions": {
    "url": "https://api.github.com/repos/shap/shap/issues/720/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/shap/shap/issues/720/timeline",
  "performed_via_github_app": null,
  "state_reason": null
}
